packages/core/src/repository.ts

```diff
@@ -0,0 +1,46 @@
+import Document from './document';
+import DocID from '@ceramicnetwork/docid';
+
+export class Repository {
+  readonly #map: Map<string, Document> = new Map();
+
+  /**
+   * Stub for async loading of the document.
+   */
+  async get(docId: DocID): Promise<Document> {
+    const found = this.#map.get(docId.toString());
+    if (found) {
+      return found;
+    } else {
+      throw new Error(`No document found for id ${docId}`);
```

Comment:
I'm concerned about the paradigm of always calling `Repository.has()` before `Repository.get()` with the assumption that if `.has()` returns true then the doc will still be there when calling `.get()`.  If the document is at the very end of the LRU cache about to age out, we call `.has()`, it returns true, then the schedule switches contexts and a new doc is inserted into the cache, then when `.get()` is schedule the document can be missing, despite having previously returned true for `.has()`.

Reply:
I guess if the call to `.has()` queries the cache and marks the doc as accessed it will reset its order in the LRU, so maybe this is unlikely to ever be an actual problem in practice.  Nevertheless I wonder is there actually value to having this separation between `has` and `get`?  Can't `get()` just return `null` if it doesn't have it?